File: checkBigchangeConnection.js
#!/usr/bin/env node
"use strict";

const {
  BigChangeClient,
  configFromEnv,
  formatResponseForDisplay,
  isSuccessResponse,
  loadDotEnv,
} = require("./bigchangeClient");

async function main() {
  loadDotEnv();

  let result;
  try {
    const client = new BigChangeClient(configFromEnv());
    result = await client.call("listmethods");
  } catch (error) {
    if (error && error.name === "BigChangeConfigError") {
      console.error(`Configuration error: ${error.message}`);
      console.error("Create a .env file using .env.example as a template.");
      return 2;
    }

    console.error(`Connection test failed: ${error.message}`);
    return 1;
  }

  console.log("Connection succeeded. BigChange returned:");
  console.log(formatResponseForDisplay(result));
  return isSuccessResponse(result) ? 0 : 1;
}

main()
  .then((exitCode) => {
    process.exitCode = exitCode;
  })
  .catch((error) => {
    console.error(`Unexpected error: ${error.stack || error.message}`);
    process.exitCode = 1;
  });

File: check_bigchange_connection.py
"""Run a safe BigChange connection test.

This script performs a read-only `listmethods` call, which the JobWatch web
service documentation describes as a way to list available service methods.
"""

from __future__ import annotations

import sys

from bigchange_client import (
    BigChangeClient,
    BigChangeConfig,
    BigChangeConfigError,
    format_response_for_display,
    is_success_response,
    load_dotenv,
)


def main() -> int:
    load_dotenv()

    try:
        client = BigChangeClient(BigChangeConfig.from_env())
        result = client.call("listmethods")
    except BigChangeConfigError as exc:
        print(f"Configuration error: {exc}", file=sys.stderr)
        print("Create a .env file using .env.example as a template.", file=sys.stderr)
        return 2
    except RuntimeError as exc:
        print(f"Connection test failed: {exc}", file=sys.stderr)
        return 1

    print("Connection succeeded. BigChange returned:")
    print(format_response_for_display(result))
    return 0 if is_success_response(result) else 1


if __name__ == "__main__":
    raise SystemExit(main())
